### DISCUSSION TITLE
In .NET8 Blazor WebApp can MainLayout.cs be interactive without a global render mode in App.razor?

### DISCUSSION THREAD

**Discussion by Ghevi** [dotnet/aspnetcore#51491] [Q&A] [answered]
Just asking for a clarification, it seems to me that if you set a render mode in the `App.razor` file to the `Routes` component so that is the same for all the pages, the `MainLayout.cs` is interactive. But if you set the render mode individually per page, then it stops being interactive and you can't set a render mode on it or you get this error:
`
InvalidOperationException: Cannot pass the parameter 'Body' to component 'MainLayout' with rendermode 'InteractiveServerRenderMode'. This is because the parameter is of the delegate type 'Microsoft.AspNetCore.Components.RenderFragment', which is arbitrary code and cannot be serialized.`

**Comment by Aquaritek** [comment, marked as answer]
The only way to make anything that Consumes a RenderFragment interactive is to wrap the component in an interactive component to let interactivity cascade. 

That said the only way to wrap MainLayout is to globally set the interactivity rendermode in the App.razor file on both the HeadOutlet component and the Routes component. 

The caveat here is that you've effectively locked yourself into the rendermode permanently for all cascading components or rather the entire app.
